json/src/main/java/org/teavm/flavour/json/emit/JsonDeserializerEmitter.java

```diff
@@ -498,11 +498,27 @@ private void emitField(PropertyInformation property, Value<Object> target,
         Type type = javaField.getGenericType();
 
         String propertyName = property.outputName;
-        Value<Node> jsonValue = emit(() -> node.get().get(propertyName));
+        Value<Node> jsonValue = readProperty(property, node, propertyName);
         Value<Object> value = convert(jsonValue, context, type, field);
         emit(() -> field.set(target.get(), value.get()));
```

Comment:
I guess you should check if property presents in JSON and only if it does, call `field.set`.

Reply:
done!